Lib/os.py

```diff
@@ -340,89 +340,97 @@ def walk(top, topdown=True, onerror=None, followlinks=False):
 
     """
     sys.audit("os.walk", top, topdown, onerror, followlinks)
-    return _walk(fspath(top), topdown, onerror, followlinks)
-
-def _walk(top, topdown, onerror, followlinks):
-    dirs = []
-    nondirs = []
-    walk_dirs = []
-
-    # We may not have read permission for top, in which case we can't
-    # get a list of the files the directory contains.  os.walk
-    # always suppressed the exception then, rather than blow up for a
-    # minor reason when (say) a thousand readable directories are still
-    # left to visit.  That logic is copied here.
-    try:
-        # Note that scandir is global in this module due
-        # to earlier import-*.
-        scandir_it = scandir(top)
-    except OSError as error:
-        if onerror is not None:
-            onerror(error)
-        return
 
-    with scandir_it:
-        while True:
-            try:
+    stack = [(False, fspath(top))]
+    while stack:
+        must_yield, top = stack.pop()
+        if must_yield:
+            yield top
+            continue
+
+        dirs = []
+        nondirs = []
+        walk_dirs = []
+
+        # We may not have read permission for top, in which case we can't
+        # get a list of the files the directory contains.  os.walk
+        # always suppressed the exception then, rather than blow up for a
+        # minor reason when (say) a thousand readable directories are still
+        # left to visit.  That logic is copied here.
```

Comment:
```suggestion
        # get a list of the files the directory contains.
        # We suppressed the exception here, rather than blow up for a
        # minor reason when (say) a thousand readable directories are still
        # left to visit.
```
Strange to say it is "copied" when this *is* the implementation of `os.walk`.

Reply:
Yeah, I mostly left comments as they were. Looks like this comment was copied from `pathlib.Path.walk` at some point